Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,450 @@
+            BeforeAll {
+                  $repoWithPR = Get-GitHubRepository -OrganizationName 'super-fake-not-real-org' |
+                      Where-Object { $_ | Get-GitHubPullRequest } |
+                      Select-Object -First 1
+                $pullRequest = $repoWithPR | Get-GitHubPullRequest | Select-Object -First 1
+                $newGitHubCodespaceParms = @{
+                    PullRequest = $pullRequest.number
+                    RepositoryId = $repoWithPR.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repoWithPR.name
+                $codespace.idle_timeout_minutes | Should -Be 30
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.pulls_url | Should -Be $pullRequest.url
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with all possible settings' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    # ClientIp = 'TODO ???? - should be instead of rather than in addition to Location, perhaps add some param validation to the function'
+                    # Devcontainer = 'Will add to test in the future when Get-GitHubDevContainer is implemented and the test repo includes one'
+                    DisplayName = 'PowerShellForGitHub pester test'
+                    Location = 'WestUs2'
+                    Machine = 'basicLinux32gb'
+                    NoMultipleRepoPermissions = $true # Not sure how to assert this, but this proves it accepts the switch without error
+                    RetentionPeriod = 10
+                    Timeout = 5
+                    # WorkingDirectory = 'TODO ???? - not sure how to handle this'
+                }
+                $codespace = $repo | New-GitHubCodespace @newGitHubCodespaceParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                # $codespace.devcontainer_path | Should -Be
+                $codespace.display_name | Should -Be $newGitHubCodespaceParms.DisplayName
+                $codespace.idle_timeout_minutes | Should -Be 5
```

Comment:
Here and elsewhere, I'd suggest that you reference the value from `$newGitHubCodespaceParams` to make this robust to changes in the future and avoid magic numbers.  So, instead of saying `5`, say:

```suggestion
                $codespace.idle_timeout_minutes | Should -Be $newGitHubCodespaceParams.Timeout
```